meetings/2020-06/june-4.md

```diff
@@ -0,0 +1,934 @@
+
+CM: Without immediately taking a position on whether I think this proposal is a good idea or not, I wanted to make sure that this issue is on your radar: Is the semantics of this intended to be the result of this operator -1, 1, 0 or is it <0, 0, or >0? The reason I raise that question is because the classic comparator that you’d use for sort to compare two integers is just to subtract them. And that’s very convenient & easy but if you have this operator in the language and one implementation / engine implemented comparison by doing subtraction and another one implemented comparison by returning +1 or -1, you now have a potential source of undefined behavior across implementations if this is not nailed down in the spec. On the other hand, anchoring how the result is to be encoded also introduces potential implementation problems. So I just wanted to make sure that this potential source of nondeterminism and undefined behavior is addressed in whatever you come up with going forward.
+
+KKL: CM anticipated half of what I was going to say. My position is similar to RBN. I find it interesting to come up with protocol for comparison, starting with Array, which is the only sensible anchor among primitives.
+Which is I think the only sensible anchor among the primitives, but as for a spaceship operator I don’t see the necessity, because for sensible comparisons like number-to-number, like CM mentioned, subtraction is a sufficient implementation.
+
+JHD: there is no way that I couldn’t see the spaceship operator to not work on numbers via subtraction. And we were very careful to show in the examples that it’s not -1, 0, or +1, it’s a negative number, 0, or a positive number, for exactly this reason.
+
+WH: Subtraction doesn’t work, it doesn’t always give you the correct result when comparing Numbers.
+
+JHD: for finite numbers perhaps
+
+WH: For finite numbers it works but if you include infinities it doesn’t.  For finite numbers, you might get an overflow. However, `+Infinity == +Infinity` is true, but `+Infinity - +Infinity` gives you a NaN.
+
+JHD: So yeah, we’d have to handle the infinities and the NaNs, just like all the Math operations in the spec, but the rest of it would be mostly subtraction and that’s the sort of thing we’d have to handle in stage 2.
+
+RPR: Queue is empty
+
+JHD: Sounds like there is pretty strong opposition against spending committee time exploring generic comparison, but people are roughly okay with addressing the problem of comparing arrays. I am unclear how we can compare arrays without requiring a comparator without also addressing generic comparison of values. But either way it seems like this problem space has even more to talk about even though we’ve been given strong feedback on which parts to focus on and which parts to avoid. So it seems worthy of exploring further in stage 1. Can we have stage 1 for the proposal given that all the strong feedback we’ve received would be weighted highly?
```

Comment:
@ljharb Thomas Levy says:

> i think Jordan said the opposite, I know i Meant the opposite. Spaceship operator implies a partial ordering of the values it operates over.

Reply:
line 514 seems correct to me; can you help me understand the confusion?

Reply:
If anything: _opposition against_ is confusing as a double negative. Maybe this would clear out the whole paragraph?

```suggestion
JHD: Sounds like there is pretty strong opposition to spend committee time exploring generic comparison, but people are roughly okay with addressing the problem of comparing arrays. I am unclear how we can compare arrays without requiring a comparator without also addressing generic comparison of values. But either way it seems like this problem space has even more to talk about even though we’ve been given strong feedback on which parts to focus on and which parts to avoid. So it seems worthy of exploring further in stage 1. Can we have stage 1 for the proposal given that all the strong feedback we’ve received would be weighted highly?
```

Reply:
sure, "opposition to" is a clearer synonym here.